### PR TITLE
add cleanliness target and better clippy coverage

### PR DESCRIPTION
**Pull request by bluejekyll**

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -256,15 +256,38 @@ description = "Touch all Rust files (for clippy)"
...
+[tasks.clippy-all-features]
+description = "Run the clippy linter with all-features"
+workspace = false
+env = { FEATURES = "--all-features" }
+run_task = { name = "clippy-inner" }
+
+[tasks.clippy]
+description = "Run the clippy linter with default features, no-default-features, and all-features"
+workspace = false
+dependencies = ["touch_all_rs"]
```
Comment (djc) [inline Makefile.toml:289]: I think the `touch_all_rs` is no longer necessary starting with 1.52 or so?

**Review by djc — COMMENTED** [review]
What is the value you see from running `clippy` with anything but `--all-features`? Assuming that default features resolves to a subset of all features, I'm not sure how this is useful.

File: Makefile.toml (hunk 1 above)
Comment (bluejekyll) [reply Makefile.toml:289]: Is that true? I wasn’t aware that was fixed. Across features it’s not necessary.

**Comment by bluejekyll** [comment]
mainly this is to catch things like unused imports on default and no-default depenedencies. I think we were allowing some warnings through based on that.

**Review by djc — APPROVED** [review]

**Pull request merged**